src/Swashbuckle.AspNetCore.Newtonsoft/SchemaGenerator/NewtonsoftDataContractResolver.cs

```diff
@@ -90,6 +90,15 @@ public DataContract GetDataContractForType(Type type)
                     keys = enumValuesAsJson.Any(json => json.StartsWith("\""))
                         ? enumValuesAsJson.Select(json => json.Replace("\"", string.Empty))
                         : keyType.GetEnumNames();
+
+                    keys = keyType.GetEnumValues().Cast<object>().Select(v =>
+                    {
+                        var dic = new Dictionary<object, int>() { { v, 0 } };
+                        var serialized = JsonConvert.SerializeObject(dic, _serializerSettings);
+                        var deserialized =
+                            JsonConvert.DeserializeObject<Dictionary<string, int>>(serialized, _serializerSettings);
+                        return deserialized.Keys.Single();
+                    });
```

Comment:
This completely replaces the code above, and (at least locally) passes all tests. I'm not a huge fan of actually doing a serialization roundtrip, but it's the only way I came up with to make sure all rules and transformations of Newtonsoft.Json are followed. Ideas welcome.

Reply:
I think that this error also happens for the STJ serializer

Reply:
Added